docs/framework/tutorial/code/cc-ingester/index.js

```diff
@@ -0,0 +1,177 @@
+            .map(function () { return $(this).attr('content'); })
+            .get();
+        const tags = wpTags.map(t => `tag:${t}`);
+        tags.unshift(wpCategory);
+
+        const dom = JSDOM.jsdom($.html(), {
+            features: {ProcessExternalResources: false},
+        });
+        const facts = rules.against(dom);
+        const html = facts.get('content')
+            .filter(fnode => fnode.scoreFor('paragraphish') > 0)
+            .map(fnode => fnode.element.outerHTML).join('');
+
+        // Load the DOM back into Cheerio
+        $ = Cheerio.load('<article>');
+        $('article').append(html);
+
+        // Do some extra cleanup to minimize the size
+        const all = $('*');
+        all.removeAttr('class');
+        all.removeAttr('style');
+        const imgs = $('img');
+        ['attachment-id', 'permalink', 'orig-file', 'orig-size', 'image-meta',
+            'comments-opened', 'medium-file', 'large-file']
+            .forEach(data => imgs.removeAttr(`data-${data}`));
+        imgs.removeAttr('srcset');  // For simplicity, only use one size
+
+        const postAsset = new Libingester.BlogArticle();
+        postAsset.set_title(title);
+        postAsset.set_synopsis(synopsis);
+        postAsset.set_canonical_uri(link);
+        postAsset.set_last_modified_date(new Date(lastModified));
+        postAsset.set_date_published(date);
+        postAsset.set_license('CC BY 4.0 International');
+        postAsset.set_author(author);
+        postAsset.set_read_more_text(`"${title}" by ${author}, used under CC BY 4.0 International / Reformatted from original`);
+        postAsset.set_tags(tags);
+
+        const thumbnailAsset = Libingester.util.download_image(imageURI);
+        hatch.save_asset(thumbnailAsset);
+        postAsset.set_thumbnail(thumbnailAsset);
+
+        // Pick out a "main image": the first <figure>
+        const figures = $('figure');
+        if (figures.length) {
+            const main = figures.first();
+            const img = $('img', main);
+            const mainImageAsset = Libingester.util.download_img(img, baseURI);
+            hatch.save_asset(mainImageAsset);
+
+            postAsset.set_main_image(mainImageAsset);
+            postAsset.set_main_image_caption($('figcaption', main).text());
+
+            $(main).remove();
+        }
+
+        // Save assets for any remaining <figure>s
+        $('figure').each(function () {
+            const img = $('img', this);
+            const figureAsset = Libingester.util.download_img(img, baseURI);
+            hatch.save_asset(figureAsset);
+        });
+
+        postAsset.set_body($);
+        postAsset.render();
+
+        hatch.save_asset(postAsset);
+    });
+}
+
+function main() {
+    const hatch = new Libingester.Hatch('cc-blog', 'en');
+    const paginator = Libingester.util.create_wordpress_paginator(feedURI);
+    Libingester.util.fetch_rss_entries(paginator, Infinity, 90)
```

Comment:
I mentioned those variables in the API docs, but I want to keep nightly ingestion out of this as much as possible, because it's out of scope for the tutorial, and it's something that external users of libingester will have to solve in their own way without SOMA. I'll add a note about the environment variables and what they can be used for, but I think during development it's much easier to edit the values in the code.

Reply:
I am not sure about editing the code being much easier than just running the ingester with an env variable set... But sounds good, as long as we do mention the variables as an option.